Tests/PromotedCoreTests/XrayTests.swift

```diff
@@ -81,8 +79,55 @@ final class XrayTests: ModuleTestCase {
     XCTAssertEqual(50, call2.timeSpent.millis)
     XCTAssertEqual(impression, call2.messages[0] as! Event_Impression)
   }
-  
+
+  func testBatchSummaries() {
+    config.xrayLevel = .batchSummaries
+    xray = Xray(deps: module)
+
+    clock.advance(toMillis: 0)
+    let actionContext = Context.function("logAction")
+    xray.executionWillStart(context: actionContext)
+    var action = Event_Action()
+    action.actionID = "fake-action-id"
+    xray.execution(context: actionContext, willLogMessage: action)
+    clock.advance(toMillis: 123)
+    xray.executionDidEnd(context: actionContext)
+
+    clock.advance(toMillis: 456)
+    xray.executionWillStart(context: .batch)
+    var logRequest = Event_LogRequest()
+    logRequest.action.append(action)
+    xray.execution(context: .batch, willLogMessage: logRequest)
+    let data = "fake http body".data(using: .utf8)!
+    xray.execution(context: .batch, willLogData: data)
+    clock.advance(toMillis: 789)
+    xray.executionDidEnd(context: .batch)
+    clock.advance(toMillis: 1234)
+    xray.executionWillStart(context: .batchResponse)
+    xray.executionDidLog(context: .batchResponse)
+    xray.execution(context: .batch, willLogData: data)
+    xray.executionDidEnd(context: .batchResponse)
+
+    let networkTime: TimeIntervalMillis = 1234
+    let batchTime: TimeIntervalMillis = 789 - 456
+
+    XCTAssertGreaterThan(xray.totalBytesSent, 0)
+    XCTAssertEqual(1, xray.batchesSentSuccessfully)
+    XCTAssertEqual(batchTime, xray.totalTimeSpent.millis)
+
+    XCTAssertEqual(1, xray.networkBatches.count)
+    let batch = xray.networkBatches[0]
+    XCTAssertGreaterThan(batch.messageSizeBytes, 0)
+    XCTAssertEqual(networkTime, batch.networkEndTime.millis)
+    // When xrayLevel == .batchSummaries,
+    // timeSpentAcrossCalls doesn't include call time.
+    XCTAssertEqual(batchTime, batch.timeSpentAcrossCalls.millis)
+    XCTAssertEqual(0, batch.calls.count)
```

Comment:
is it worth asserting any other fields?

Reply:
Most of the fields are being checked in the `testSingleBatch` test. This test performs fewer checks because we're expecting less data to be filled out (ie. `XCTAssertEqual([], batch.calls)`)